### ISSUE TITLE
ES5 component size

### ISSUE THREAD

**Issue by oravecz** [basic-web-components/basic-web-components#27]
Looking at the demo that uses three components, the minified size of the ES5 components is greater than 500K, and the gzip size is around 124K for three components and dependencies. This excludes the polyfills. 

Does the way these components are packaged for NPM make it difficult to reduce the repetition of mixins that are shared across components?

**Comment by JanMiksovsky** [comment]
Can you confirm which demo you're looking at? I'd like to make sure I'm looking at the same thing.

**Comment by oravecz** [comment]
Yes, I was looking at the `dist` folder in `packages/demos/dist` for this repo.

**Comment by JanMiksovsky** [comment]
Okay, thanks.

We're not yet minifying our builds yet. For now, they're just the transpiled output of Babel + Browserify. @robbear has been working on adding real minification to our work, at which point the build files will get substantially smaller. While this work is going on, you could of course apply your own minifier.

E.g., running Uglify on dist/demos.js and then gzip gives the sizes:

```
$ du -h -d 1 demos.min.*
 84K	demos.min.js
 12K	demos.min.js.gz
```

I should also mention that our carousel component is the most complex of all the ones the project has built to date, and is consequently the largest. A smaller component like `basic-autosize-textarea` is perhaps a third the size of the carousel.

**Comment by oravecz** [comment]
Thanks. In addition to running minify are you also doing something to remove the inclusion of babel's helper functions (e.g. _`createClass()`) that gets included once for each file when you minify?

**Comment by robbear** [comment]
We'll look into this soon, but as an intermediary step, we'll probably go with the helpers being inlined redundantly for now and then revisit.
